fast_matched_filter/fast_matched_filter.py

```diff
@@ -92,11 +92,30 @@ def matched_filter(templates, moveouts, weights, data, step, arch='cpu'):
         print("Compiled library for {} not loaded; exiting!".format(arch))
         return
 
-    n_templates = np.int32(templates.shape[0])
-    n_stations = np.int32(data.shape[0])
-    n_components = np.int32(data.shape[1])
+    if templates.ndim == 2:
```

Comment:
Allowing this flexibility seems kinda dangerous to me - if I were you I would just have some catch like:
```python
assert templates.ndim > 2
```
mostly to avoid any confusion. If someone wants to pass just one template then it should still be formatted in the same way.

Reply:
I see the wisdom in what you're saying, but I'm also very frustrated by python eliminating dimensions when you index matrices.

That combined with the fact that the underlying C code is completely agnostic about what format you feed to it, I feel like there's no reason to force a [station x component x time] format.

I suppose my question would be: is my implementation safe? Once you determine n_templates and look at format of data, I _think_ you have all of the ingredients you need to reshape things into a standard format.

But is this correct?